src/lda.jl

```diff
@@ -60,15 +60,15 @@ end
 
 #### Multiclass LDA Stats
 
-mutable struct MulticlassLDAStats{T<:Real}
+mutable struct MulticlassLDAStats{T<:Real, M<:AbstractMatrix{T}, N<:AbstractMatrix{T}}
     dim::Int              # sample dimensions
     nclasses::Int         # number of classes
     cweights::Vector{T}   # class weights
     tweight::T            # total sample weight
     mean::Vector{T}       # overall sample mean
     cmeans::Matrix{T}     # class-specific means
-    Sw::Matrix{T}         # within-class scatter matrix
-    Sb::Matrix{T}         # between-class scatter matrix
+    Sw::M                 # within-class scatter matrix
+    Sb::N                 # between-class scatter matrix
```

Comment:
Why use type parameter here?  We do not expect `Sw` and `Sb` to have different types, so  `AbstractMatrix{T}` for both of them is ok.

Reply:
`Sw` and `Sb` should have the same element types but in general they may not have the same Container type. 
For my use case I am estimating `Sw` using  `LinearShrinkage` estimator from **CovarianceEstimation.jl** which returns an `LinearAlgebra.Symmetric` object and estimating `Sb` using `SimpleCovariance` from **StatsBase.jl** which returns a `Array`.
The only way to make this work is if if use two Container types `M` and `N` as shown above. (`N` and `M` have the elements of type `T`) plus to improve performance i avoided the use of abstract types in the struct def